### ISSUE TITLE
A11y_Windows Terminal_Windows PoerShell_Screen reader: User is not able to navigate to the content (text) present when screen reader is on even in scan mode

### ISSUE THREAD

**Issue by Amulya-95** [microsoft/terminal#5422]
GitHubTags: #A11y_WindowsTerminal; #WindowsTerminal; #A11yTCS; #A11yMAS; #Win10; #DesktopApp; #ScreenReader; #Narrator; #MAS1.3.1; #C+E-H; #A11ySev1;

[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-tagott%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458847261333&sdata=PJMiKjb64uwhBN98LZpctnBiAVeOvw%2BLBrnxBtrjl2M%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier

**Note:** Similar Issue is observed in 'Command prompt' and 'Azure cloud shell'

**Environment Details:**
Windows Terminal
Version: 0.10.781.0
OS: Windows 10 (Version 1909, Build: 18363.778)

**Repro Steps:**
1.Turn on the screen reader
2.Hit the URL "aka.ms/terminal" and install "Windows terminal (Preview)" application
3.Open "Windows terminal (Preview)" application
4.Tab to "Windows Powershell" tab 
5.Type "LS" and press enter
6.Verify whether user is able to navigate through and screen reader is reading the content (text) present

**Actual Result:**
User is not able to navigate to the content (text) present when screen reader is on even in scan mode

**Expected Result:**
User should be able to navigate to the content (text) present when screen reader should read the content

**User Impact:**
Visually disabled users will find difficulty, if the screen reader is not reading the content (text) present

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

**Attachment**
[Bug_User is not able to navigate to the content present and screen reader is not reading it.pptx](https://github.com/microsoft/terminal/files/4503523/Bug_User.is.not.able.to.navigate.to.the.content.present.and.screen.reader.is.not.reading.it.pptx)

[Bug_NVDA is not reading the content.pptx](https://github.com/microsoft/terminal/files/4503827/Bug_NVDA.is.not.reading.the.content.pptx)

**Comment by Frnaklin12** [comment]
thats a big bug

**Comment by Amulya-95** [comment]
Regressed the bug in latest environment provided. Still able to reproduce the issue.
Attached the attachments below for reference.

[[Not Fixed]Bug 5422_not able to navigate to the content (text) even in scan mode.pptx](https://github.com/microsoft/terminal/files/4610085/Not.Fixed.Bug.5422_not.able.to.navigate.to.the.content.text.even.in.scan.mode.pptx)

[[Not Fixed]Bug 5422_NVDA_still able reproduce the issue.pptx](https://github.com/microsoft/terminal/files/4610180/Not.Fixed.Bug.5422_NVDA_still.able.reproduce.the.issue.pptx)

**Comment by carlos-zamora** [comment]
## Narrator
Scan Mode should be working on Narrator. However, new output is not detected by Narrator due to a bug on Narrator's side.

Your attached video shows that new output is not detected. However, it seems like you entered scan mode, then moved focus to another UI element (thus, exiting scan mode). If you keep your focus on the terminal's text buffer, then enter scan mode (<kbd>WinKey+ctrl+enter</kbd>), you can verify that the following navigation modes work:
- character: left/right arrow
- word: ctrl+left/right arrow
- line: ctrl+up/down arrow

Just a guess, but I'm guessing you moved focus to another UI element on accident by pressing the up/down arrow, which is by design.

## NVDA
Please be sure that you're using the NVDA v2020.1 or later (which you can find on their releases page). Also make sure that you have it configured properly:
- in NVDA Settings --> Advanced, enable "Use UI Automation to access the Windows Console when available" under "Microsoft UI Automation"

That said, again, I do not see this issue on my end. NVDA is verified to be working with new output and their equivalent scan mode.

Your attached video states that you are trying to navigate the buffer using the arrow keys. In desktop mode, NVDA works best with a numpad (numlock off), where the following navigation can be performed:
- character:
   - 1 --> read previous character
   - 2 --> read current character
   - 3 --> read next character
- word:
   - 4 --> read previous word
   - 5 --> read current word
   - 6 --> read next word
- line: 
   - 7 --> read previous line
   - 8 --> read current line
   - 9 --> read next line

Using the arrow keys should go directly to the shell to allow you to navigate the shell's history or move the cursor within the prompt as you would normally. This is by design.

## Follow-up
Please confirm that you are configured properly as stated above and you are trying to interact with Windows Terminal as mentioned above. If not, how are you trying to interact with Terminal in these situations?

**Comment by Amulya-95** [comment]
Regressed the bug. Followed above procedure mentioned in above comments and was not able to reproduce the issue.  When used the numpad (turnoff Numlock), NVDA was reading the content and Narrator as well reads the content in scan mode. Hence closing the bug.

Please find the attachment for reference.
[Bug 5422_NVDA is reading the content.pptx](https://github.com/microsoft/terminal/files/4649904/Bug.5422_NVDA.is.reading.the.content.pptx)
[Bug 5422_Narrator is reading the content.pptx](https://github.com/microsoft/terminal/files/4649907/Bug.5422_Narrator.is.reading.the.content.pptx)
